### PR TITLE
Feature/RPA-805 Pomijanie procesów gości

### PR DESCRIPTION
**Pull request by star-in-the-jar**
## Description
Now, processes created by guests are not visible for normal users (are visible only for admins).

The guests' processes are no longer visible:
- on processes list page (the grid), 
- through searchbar,
- in action StartProcess.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/domain/UserAuthority.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private String authorityName;
+
+    public Long getUserId() {
+        return userId;
+    }
+
+    public void setUserId(Long userId) {
+        this.userId = userId;
+    }
+}
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/domain/UserAuthority.java]: the linking tables should not exist as domains. Create Many-to-Many field in the Authority entity and link there users just like we do with feature keys

Comment (preludium) [reply runbotics-orchestrator/src/main/java/com/runbotics/domain/UserAuthority.java]: then just get all guests authorities and you have all users joined to it

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java [hunk 2]
```diff
@@ -192,4 +200,21 @@ private Specification<Process> isCreatedByUser() {
...
+        List<ProcessDTO> filteredProcesses = new ArrayList<>();
+
+        if(!hasRequesterRoleAdmin) {
+            for (ProcessDTO process : processes) {
+                if(!guestIds.contains(process.getCreatedBy().getId())) {
+                    filteredProcesses.add(process);
+                }
+            }
+            return filteredProcesses;
+        }
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java]: all of this you can replace with a single stream chain

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java]: DONE

**Review by star-in-the-jar — COMMENTED** [review]
self reviewed

File: runbotics-orchestrator/src/main/java/com/runbotics/domain/UserAuthority.java (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics-orchestrator/src/main/java/com/runbotics/domain/UserAuthority.java]: done

**Review by preludium — APPROVED** [review]

**Pull request merged**